readthedocs/oauth/services/gitlab.py

```diff
@@ -0,0 +1,237 @@
+            repo.account = self.account
+            owner = fields.get('owner') or {}
+            repo.avatar_url = (fields.get('avatar_url') or
+                               owner.get('avatar_url') or
+                               self.default_avatar['repo'])
+            repo.json = json.dumps(fields)
+            repo.save()
+            return repo
+        else:
+            log.info(
+                u'Not importing {0} because mismatched type: public={1}'.format(
+                    fields['name_with_namespace'],
+                    fields['public'],
+                )
+            )
+
+    def create_organization(self, fields):
+        """Update or create remote organization from GitLab API response
+
+        :param fields: dictionary response of data from API
+        :rtype: RemoteOrganization
+        """
+        try:
+            organization = RemoteOrganization.objects.get(
+                slug=fields.get('path'),
+                users=self.user,
+                account=self.account,
+            )
+        except RemoteOrganization.DoesNotExist:
+            organization = RemoteOrganization.objects.create(
+                slug=fields.get('path'),
+                account=self.account,
+            )
+            organization.users.add(self.user)
+
+        organization.name = fields.get('name')
+        organization.account = self.account
+        organization.url = u'{url}/{path}'.format(
+            url=self.adapter.provider_base_url, path=fields.get('path')
+        )
+        avatar = fields.get('avatar') or {}
+        if avatar.get('url'):
+            organization.avatar_url = u'{url}/{avatar}'.format(
+                url=self.adapter.provider_base_url,
+                avatar=avatar.get('url'),
+            )
+        else:
+            organization.avatar_url = self.default_avatar['org']
+        organization.json = json.dumps(fields)
+        organization.save()
+        return organization
+
+    def setup_webhook(self, project):
+        """Set up GitLab project webhook for project
+
+        :param project: project to set up webhook for
+        :type project: Project
+        :returns: boolean based on webhook set up success
+        :rtype: bool
+        """
+        session = self.get_session()
+        resp = None
+        repositories = RemoteRepository.objects.filter(clone_url=project.vcs_repo().repo_url)
+
+        if not repositories.exists():
+            log.error('GitLab remote repository not found')
+            return False, resp
+
+        repo_id = repositories[0].get_serialized()['id']
+        # See: http://doc.gitlab.com/ce/api/projects.html#add-project-hook
+        data = json.dumps({
+            'id': repo_id,
+            'push_events': True,
+            'issues_events': False,
+            'merge_requests_events': False,
+            'note_events': False,
+            'tag_push_events': True,
+            'url': u'https://{0}/gitlab'.format(settings.PRODUCTION_DOMAIN),
```

Comment:
In #2433 these webhook endpoints were deprecated. A new handler for GitLab will be required and this should be updated. We can either address this in the PR, or I'm also +1 on filing a new PR to address this change.